pyanaconda/modules/payloads/source/url/url.py

```diff
@@ -0,0 +1,145 @@
+#
+# Kickstart module for URL payload source.
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+from pyanaconda.core.constants import BASE_REPO_NAME
+from pyanaconda.core.signal import Signal
+from pyanaconda.core.util import ProxyString, ProxyStringError
+from pyanaconda.modules.common.errors import InvalidValueError
+from pyanaconda.modules.common.structures.payload import RepoConfigurationData
+from pyanaconda.modules.payloads.constants import SourceType, URLType
+from pyanaconda.modules.payloads.source.source_base import PayloadSourceBase
+from pyanaconda.modules.payloads.source.url.url_interface import URLSourceInterface
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class URLSourceModule(PayloadSourceBase):
+    """The URL source payload module."""
+
+    REPO_NAME_ID = 0
+
+    def __init__(self):
+        super().__init__()
+        self._repo_configuration = RepoConfigurationData()
+        self.repo_configuration_changed = Signal()
+
+        self._url_source_id = self._generate_source_id()
+        self._set_repo_config_name()
+
+        self._install_repo_enabled = False
+        self.install_repo_enabled_changed = Signal()
+
+    @classmethod
+    def _generate_source_id(cls):
+        source_id = cls.REPO_NAME_ID
+        cls.REPO_NAME_ID = cls.REPO_NAME_ID + 1
+
+        return source_id
```

Comment:
Btw. I would generate here the full repo name. The source ID is not really interesting.

Reply:
Fixed.